Tools/Scripts/webkitpy/port/visionos.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2024 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import logging
+
+from webkitcorepy import Version
+
+from webkitpy.common.version_name_map import VersionNameMap, INTERNAL_TABLE
+from webkitpy.port.config import apple_additions
+from webkitpy.port.device_port import DevicePort
+from webkitpy.xcode.device_type import DeviceType
+
+
+_log = logging.getLogger(__name__)
+
+
+class VisionOSPort(DevicePort):
+    port_name = 'visionos'
+
+    CURRENT_VERSION = Version(1)
+    DEVICE_TYPE = DeviceType(software_variant='visionOS')
+
+    def __init__(self, *args, **kwargs):
+        super(VisionOSPort, self).__init__(*args, **kwargs)
+
+        if not self.get_option('webkit_test_runner', False):
+            raise RuntimeError('DumpRenderTree is not supported on this platform.')
+
+    def driver_name(self):
+        if self.get_option('driver_name'):
+            return self.get_option('driver_name')
+        return 'WebKitTestRunnerApp.app'
+
+    def version_name(self):
+        if self._os_version is None:
+            return None
+        return VersionNameMap.map(self.host.platform).to_name(self._os_version, platform=VisionOSPort.port_name)
+
+    def default_baseline_search_path(self, **kwargs):
```

Comment:
Nit: can we not move this to somewhere shared between all the Apple embedded ports? I guess this is slightly complicated because we need to be able to add multiple platform names (`VisionOSPort.port_name` and `self.port_name`)?

Reply:
The issue is that while watchOS and visionOS share a baseline search path format, iOS and iPadOS don't. All of this is made further complicated by the fact that we're probably going to end up adding a UIKit baseline in short order here.

Reply:
iPadOS is the only real special-case here, right?